tensorboard/data/server/cli.rs

```diff
@@ -254,6 +272,23 @@ fn write_port_file(path: &Path, port: u16) -> std::io::Result<()> {
     Ok(())
 }
 
+/// Writes error to the given file, or to stderr as a fallback.
+fn write_startup_error(path: Option<&Path>, error: dynamic_logdir::Error) {
```

Comment:
Optional and feel free to tell me this is less idiomatic but I find the `match` here a little more complicated to think about than the relatively more prosaic nested if with fall-through:

```
if let Some(p) = path {
  let write_to_file = ...
  if write_to_file(p).is_ok() {
    return;
  }
}
eprintln!("fatal: {}", error);
```